File: jimmer-ddd-java-sample/domain/src/main/java/org/morecup/jimmerddd/java/sample/domain/order/Aftermarket.java
package org.morecup.jimmerddd.java.sample.domain.order;

import org.babyfish.jimmer.sql.DissociateAction;
import org.babyfish.jimmer.sql.Entity;
import org.babyfish.jimmer.sql.ManyToOne;
import org.babyfish.jimmer.sql.OnDissociate;
import org.morecup.jimmerddd.java.sample.domain.base.BaseEntity;

@Entity
public interface Aftermarket extends BaseEntity {
    String reason();

    @ManyToOne
    @OnDissociate(DissociateAction.DELETE)
    Order order();
}

File: jimmer-ddd-java-sample/domain/src/main/java/org/morecup/jimmerddd/java/sample/domain/order/Aftermarket.kt
package org.morecup.jimmerddd.kotlin.sample.domain.order

import org.babyfish.jimmer.sql.DissociateAction
import org.babyfish.jimmer.sql.Entity
import org.babyfish.jimmer.sql.ManyToOne
import org.babyfish.jimmer.sql.OnDissociate
import org.morecup.jimmerddd.kotlin.sample.domain.base.BaseEntity

@Entity
interface Aftermarket: BaseEntity {
    val reason: String

    @ManyToOne
    @OnDissociate(DissociateAction.DELETE)
    val order: Order
}